Makes EventRole Ids properly set ID names.

Vermidia opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
Event roles like ERT or shuttle roles now have their ids set their name on the card.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
So you know who the card belonged to, and so it shows up when opening the PDA.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
New component: ID bind that activates when a mind is added then removes itself.
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/32827189/0297cf19-c142-4cd6-ab74-2d5ea24ff1ab



## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->
No?
**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
no cl just flavor

Dutch-VanDerLinde commented:
why does  the moth have hte identity Felana Leander when the id card is Specialist Chameleon

Vermidia commented:
Weirdness because randomizeName isn't set to false on ERT/CBURN meaning they have their name randomized twice, which I should probably fix here.

Vermidia commented:
Moved the system to server, and changed it to activate on mind added (Couldn't find a way to make sure it only activates after it applies the name and everything except that)

metalgearsloth commented on Content.Server/Access/Systems/IdBindSystem.cs:
```diff
@@ -0,0 +1,48 @@
+        if (!TryComp<PdaComponent>(uPda, out var pDA))
+            return;
+
+        _pdaSystem.SetOwner(uPda.Value, pDA, data.EntityName);
+        //Remove after running once
+        EntityManager.RemoveComponent(ent, ent.Comp);
```
You should not be touching components in the eventbus as this will break.

metalgearsloth commented on Content.Server/Access/Components/IdBindComponent.cs:
```diff
@@ -0,0 +1,18 @@
+using Robust.Shared.GameStates;
+
+namespace Content.Server.Access.Components;
+/// <summary>
+/// Makes it so when starting gear loads up, the name on a PDA/Id (if present) is changed to the character's name.
+/// </summary>
+
+[RegisterComponent]
+public sealed partial class IdBindComponent : Component
+{
+    /// <summary>
+    /// If true, also tries to get the PDA and set the owner to the entity
+    /// </summary>
+    [DataField]
+    public bool BindPDAOwner = true;
+
+}
+
```
```suggestion
namespace Content.Server.Access.Components;

/// <summary>
/// Makes it so when starting gear loads up, the name on a PDA/Id (if present) is changed to the character's name.
/// </summary>
[RegisterComponent]
public sealed partial class IdBindComponent : Component
{
    /// <summary>
    /// If true, also tries to get the PDA and set the owner to the entity
    /// </summary>
    [DataField]
    public bool BindPdaOwner = true;
}
```

slarticodefast commented on Resources/Prototypes/Entities/Mobs/Player/humanoid.yml:
```diff
@@ -71,7 +72,6 @@
 - type: randomHumanoidSettings
   id: ERTLeader
   parent: EventHumanoidMindShielded
-  randomizeName: false
```
wouldn't this randomize the name twice?

slarticodefast commented on Resources/Prototypes/Entities/Mobs/Player/humanoid.yml:
```diff
@@ -458,6 +458,7 @@
 - type: randomHumanoidSettings
   id: CBURNAgent
   parent: EventHumanoidMindShielded
+  randomizeName: false
```
this should already be inherited from EventHumanoidMindShielded

slarticodefast reviewed (changes requested):
Sorry for the long wait with the review. We are really backlogged at the moment.

slarticodefast commented:
Oh, ancient PR finally merged. Now that I look at it again, shouldn't this change it on map init rather than when a mind is added?

Vermidia commented:
This is very old but the reason it does not is because it would result in the PDA/ID getting the first name generated instead of the final name.
